build_projects/dotnet-cli-build/GenerateGuidFromName.cs

```diff
@@ -0,0 +1,82 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Runtime.InteropServices;
+using System.Net.Http;
+using Microsoft.Build.Utilities;
+using Microsoft.DotNet.Cli.Build.Framework;
+using Microsoft.DotNet.InternalAbstractions;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System.Security.Cryptography;
+
+using static Microsoft.DotNet.Cli.Build.Framework.BuildHelpers;
+
+namespace Microsoft.DotNet.Cli.Build
+{
+    public class GenerateGuidFromName : Task
```

Comment:
In general, for shared code like this, I'd say let's either:

1. Delete the existing code, and have it call into this Task class.
 - For example, anywhere that calls Utils.GenerateGuidFromName, could just call GenerateGuidFromName.GenerateGuid.
2. Have the Task class call into the existing shared util.

That way we don't have duplication as we go.